### PR TITLE
client/tailscale,cmd/k8s-operator: set distinct User-Agents

### PR DESCRIPTION
**Pull request by oxtoacart**
This will allow us to distinguish OSS client activity from our other Go client, and will also allow us to distinguish k8s activity.

Updates tailscale/corp#23838

### CODE REVIEW

File: client/tailscale/tailscale.go [hunk 1]
```diff
@@ -110,17 +114,16 @@ func (c *Client) Do(req *http.Request) (*http.Response, error) {
...
 }
 
 // sendRequest add the authentication key to the request and sends it. It
 // receives the response and reads up to 10MB of it.
 func (c *Client) sendRequest(req *http.Request) ([]byte, *http.Response, error) {
-	if !I_Acknowledge_This_API_Is_Unstable {
-		return nil, nil, errors.New("use of Client without setting I_Acknowledge_This_API_Is_Unstable")
-	}
-	c.setAuth(req)
-	resp, err := c.httpClient().Do(req)
```
Comment (oxtoacart) [inline client/tailscale/tailscale.go:123]: Drive-by, this code duplicated what we do in `Do()`. Consolidating allowed me to add the User-Agent header without a DRY violation.

**Review by mpminardi — APPROVED** [review]
